l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -724,16 +788,15 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
             l2TxHash = BRIDGE_HUB.requestL2TransactionDirect{value: msg.value}(request);
         }
 
-        bytes32 txDataHash = keccak256(abi.encode(_prevMsgSender, _l1Token, _amount));
         // Save the deposited amount to claim funds on L1 if the deposit failed on L2
-        depositHappened[ERA_CHAIN_ID][l2TxHash] = txDataHash;
+        depositHappened[ERA_CHAIN_ID][l2TxHash] = keccak256(abi.encode(_assetId, abi.encode(_amount, _prevMsgSender))); // Tx Data Hash
```

Comment:
Are you sure the new tx hash should be used here? For instance, for claiming failed transfer of legacy token we typically require old tx hash format